operator/pkg/parsed/template_to_module.go

```diff
@@ -74,12 +74,22 @@ func TemplatesToModules(
 	return modules, nil
 }
 
+func CreateModuleName(moduleName string, kymaName string) string {
+	return moduleName + kymaName
+}
+
 func NewModule(
 	template *v1alpha1.ModuleTemplate,
 	verification signature.Verification,
 ) (*unstructured.Unstructured, error) {
-	component := &template.Spec.Data
-
+	component := template.Spec.Data.DeepCopy()
+	if template.Spec.Target == v1alpha1.TargetRemote {
+		resource := template.Spec.Data.DeepCopy()
+		if err := mergeIntoSpecResource(resource, component); err != nil {
+			return nil, err
+		}
+		component.SetKind("Manifest")
```

Comment:
Why do you set kind but not group/version?

Reply:
group version is already copied from template.Spec.Data by `resource` and  `apiVersion`, this two fields should be mandatory defined in module template, or am I wrong here?